### PR TITLE
add option to filter vectors field in 3dSDPViewer

### PR DESCRIPTION
**Pull request by kerautret**
# PR Description
Add an option to filter vectors field in 3dSDPViewer to obtain better visualization when too numerous vectors are present

# Checklist

- [x] Doxygen documentation of the code completed (classes, methods, types, members...).
- [x] Main tool doxygen documentation (following existing documentation of [DGtalTools documentation](http://dgtal.org/doc/tools/nightly/).
- [x] Check if it follows the tools structure described in [CONTRIBUTING.md](https://github.com/DGtal-team/DGtalTools/blob/master/CONTRIBUTING.md)
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtalTools/blob/master/ChangeLog.md) added.
- [x] Update the readme with potentially a screenshot of the tools if it applies. 
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).

### CODE REVIEW

File: visualisation/3dSDPViewer.cpp [hunk 1]
```diff
@@ -124,6 +124,11 @@ typedef Viewer3D<Z3i::Space, Z3i::KSpace> Viewer;
                                         determined by two consecutive point 
                                         given, each point represented by its 
                                         coordinates on a single line.
+  --filterVectors arg (=100)            filters vector input file in order to 
+                                        display only the [arg] pourcent of the
```
Comment (dcoeurjo) [inline visualisation/3dSDPViewer.cpp]: percentage

File: visualisation/3dSDPViewer.cpp [hunk 2]
```diff
@@ -124,6 +124,11 @@ typedef Viewer3D<Z3i::Space, Z3i::KSpace> Viewer;
                                         determined by two consecutive point 
                                         given, each point represented by its 
                                         coordinates on a single line.
+  --filterVectors arg (=100)            filters vector input file in order to 
+                                        display only the [arg] pourcent of the 
+                                        input vectors (uniformly selected, to 
+                                        be used with option --drawVectors else
```
Comment (dcoeurjo) [inline visualisation/3dSDPViewer.cpp]: else->otherwise

File: visualisation/3dSDPViewer.cpp [hunk 3]
```diff
@@ -212,7 +217,9 @@ int main( int argc, char** argv )
   ("lineSize",  po::value<double>()->default_value(0.2), "defines the line size (used when the --drawLines or --drawVectors option is selected). (default value 0.2))")
   ("primitive,p", po::value<std::string>()->default_value("voxel"), "set the primitive to display the set of points (can be sphere, voxel (default), or glPoints (opengl points).")
   ("drawVectors,v", po::value<std::string>(), "SDP vector file: draw a set of vectors from the given file (each vector are determined by two consecutive point given, each point represented by its coordinates on a single line.")
-  ("interactiveDisplayVoxCoords", "by using this option the pixel coordinates can be displayed after selection (shift+left click on voxel)." );
+  ("filterVectors",po::value<double>()->default_value(100.0), "filters vector input file in order to display only the [arg] pourcent of the input vectors (uniformly selected, to be used with option --drawVectors else no effect). " )
```
Comment (dcoeurjo) [inline visualisation/3dSDPViewer.cpp]: same here

File: visualisation/3dSDPViewer.cpp [hunk 4]
```diff
@@ -420,7 +427,13 @@ int main( int argc, char** argv )
       {
         trace.info()<<"Warning the two set of points doesn't contains the same number of points, some vectors will be skipped." << std::endl;
       }
-      for(unsigned int i =0; i<vectorsPt.size()-1; i=i+2)
+      int step=1;
+      if(vm.count("filterVectors"))
+        {
+          double percent = vm["filterVectors"].as<double>();
```
Comment (dcoeurjo) [inline visualisation/3dSDPViewer.cpp]: percentage

**Comment by kerautret** [comment]
thanks @dcoeurjo corrected ;)

**Comment by dcoeurjo** [comment]
thanks. merging

**Pull request merged**